src/inmanta/deploy/scheduler.py

```diff
@@ -1129,7 +1131,13 @@ async def _update_scheduler_state_for_finished_deploy(
                 # except for the event to notify dependents of failure recovery (to unblock skipped for dependencies)
                 # because we might otherwise miss the recovery (in the sense that the next deploy wouldn't be a transition
                 # from a bad to a good state, since we're transitioning to that good state now).
+
+                # The second part of the or would not be required because is implied by the first,
+                # except that we don't enforce the hash diff.
+                # We emit a warning if we observe this, but that still doesn't prevent it.
+                # While it should not happen it can
```

Comment:
Good addition, but not clear what it's attached to. Perhaps add it right above the condition?